src/main/kotlin/org/jmailen/gradle/kotlinter/tasks/lint/LintWorkerAction.kt

```diff
@@ -31,28 +30,30 @@ abstract class LintWorkerAction : WorkAction<LintWorkerParameters> {
     private val ktLintParams: KtLintParams = parameters.ktLintParams.get()
 
     override fun execute() {
-        resetEditorconfigCacheIfNeeded(
-            changedEditorconfigFiles = parameters.changedEditorconfigFiles,
+        val ktLintEngine = KtLintRuleEngine(
+            ruleProviders = resolveRuleProviders(defaultRuleSetProviders, ktLintParams.experimentalRules),
```

Comment:
❓ 
Since experimental rules can be now also enabled via the editorconfig, shouldn't we always append them here? 
```suggestion
            ruleProviders = resolveRuleProviders(defaultRuleSetProviders, true),
```
I imagine it would be surprising to the users if they enabled experimental rules via `.editorconfig`, as stated in ktlint docs, and the rules wouldn't be used out-of-the-box.

Reply:
So just to be clear, you're suggesting to change the plugin behavior for this version? I aimed at keeping compatibility and not introducing any new changes, but passing `true` here (in fact I can just remove this param whatsoever) _will_ change the behavior 🤔 I thought it was possible to enable experimental rules already before, no?

I'd wait for @jeremymailen input here, I agree with you that we should load all rules, but at the same time I feel like doing that without moving to `EditorConfigDefaults` or deprecating Kotlinter rules configuration is a half-solution and might end up being more confusing. As a user, I'd prefer to have a single release which fully revamps the way I configure the rules

Reply:
>So just to be clear, you're suggesting to change the plugin behavior for this version? 

Not really 🤔 I mean, the plugin is just a wrapper around ktlint. It avoids overriding default ktlint behavior. So if ktlint now allows enabling experimental rules via `editorconfig`, I don't see a reason why we would want to prevent that.

> I thought it was possible to enable experimental rules already before, no?

As far as I understand, before 0.48.0, the only way to enable experimental rules was to pass additional RuleSet providers, which had to be done by the plugin. Now, as a ktlint user, by adding the `ktlint_experimental=true`, I'd expect the experimental rules to be enabled.
Screenshot from their [documentation](https://pinterest.github.io/ktlint/faq/#why-is-editorconfig-property-disabled_rules-deprecated-and-how-do-i-resolve-this)
<img width="879" alt="image" src="https://user-images.githubusercontent.com/36954793/213670201-a27350cf-5870-43e6-9d55-978c27af8de8.png">


> but passing true here will change the behavior

Which behavior will change specifically? 🤔 Starting from 0.48.0 ktlint won't run experimental rules unles they are explicitly enabled, right?

Reply:
Ah I thought it was possible to enable experimental rules via `.editorconfig` before as well. So yes, pulling all rules should not be harmful, but it will still be confusing if someone enables experimental rules in Kotlinter but wants to disable them only for some directory. We'd support _some_ Ktlint configuration but not fully, it _can_ get confusing

Reply:
Done, all rules will now be passed to Ktlint

Reply:
Yeah, as mentioned above, I support removing kotlinter's configuration of rules and delegating this entirely to editorconfig and ktlint's use of it. I couple assumptions I'm making
- This harmonizes with people adding customer rule set to the build? I think so, just mentioning it
- We properly set the gradle targets as outdated when various layers of editorconfig change -- I think we got this change last time around with the enhanced ktlint APIs if I remember right.

Reply:
> This harmonizes with people adding customer rule set to the build? I think so, just mentioning it

Afaiu yes, the rules will be _added_ via Kotlinter mechanism and configured in `.editorconfig`. Though now that I think of it, I guess Ktlint might even want to load rules on the classpath itself

> We properly set the gradle targets as outdated when various layers of editorconfig change

That I don't fully understand so can't answer